cache/FIFO_FH.h

```diff
@@ -606,12 +620,13 @@ bool FIFO_FHCache<TKey, TValue, THash>::find(TValue& ac,
     // we can re-check this and remove it if possible
 
     // Generally when we found the thing in frozen cache
-    if(m_fasthash->find(key, ac) && (ac != nullptr))
+    std::unique_lock<ListMutex> lockMap(m_mapMutex);
+    if(m_fasthash->find(key, ac) && (ac != nullptr) && (chunkMapper[key] > globalCounter))
 #else
     if(m_fasthash->find(key, ac))
```

Comment:
Maybe add to this conditional as well just in case macro HANDLE_WRITE is disabled